tests/unit/test_calibration.py

```diff
@@ -0,0 +1,271 @@
+        calibrator._extract_no_motion(acceleration=dummy_measure)
+
+
+def test_extract_no_motion() -> None:
+    """Check extract no motion."""
+    date = datetime.now()
+    delta = timedelta(seconds=1)
+    dummy_datetime = [date + (i * delta) for i in range(60)]
+    sphere_check_data = np.full((60, 3), 0.31)
+    sphere_check_data[30:] = -sphere_check_data[30:]
+
+    dummy_measure = models.Measurement(
+        measurements=sphere_check_data, time=pl.Series(dummy_datetime).alias("time")
+    )
+    calibrator = calibration.Calibration()
+
+    no_motion_data = calibrator._extract_no_motion(dummy_measure)
+
+    assert no_motion_data.shape[0] > 0, "No non-motion epochs found"
+    assert np.all(no_motion_data.min(axis=0) <= -calibrator.min_acceleration)
+    assert np.all(no_motion_data.max(axis=0) >= calibrator.min_acceleration)
+
+
+def test_closest_point_fit() -> None:
+    """Test closest point fit."""
+    dummy_no_motion = np.full((100, 3), (1 / np.sqrt(3)))
+    calibrator = calibration.Calibration()
+
+    linear_transform = calibrator._closest_point_fit(dummy_no_motion)
+
+    assert np.allclose(linear_transform["scale"], 1.0)
+    assert np.allclose(linear_transform["offset"], 0.0)
+
+
+def test_calibrate_calibration_error() -> None:
+    """Test failure due to calibration error."""
+    no_motion_data = np.full((3600, 3), 1.0)
+    dummy_measure = create_dummy_measurement(
+        sampling_rate=1, duration_hours=1, all_same_num=1.0
+    )
+    calibrator = calibration.Calibration(min_error=0.0001, max_iterations=5)
+
+    with mock.patch.object(
+        calibration.Calibration, "_extract_no_motion"
+    ) as mock_extracted_data:
+        mock_extracted_data.return_value = no_motion_data
+
+        with pytest.raises(calibration.CalibrationError):
+            calibrator._calibrate(dummy_measure)
+
+
+def test_calibration_successful() -> None:
+    """Test successful calibration."""
+    no_motion_data = np.full((3600, 3), (1 / np.sqrt(3) + 1))
+    linear_transformation = {"scale": 1, "offset": -1}
+    dummy_measure = create_dummy_measurement(
+        sampling_rate=1, duration_hours=1, all_same_num=(1 / np.sqrt(3) + 1)
+    )
+
+    with mock.patch.object(
+        calibration.Calibration, "_extract_no_motion"
+    ) as mock_extracted_data:
+        with mock.patch.object(
+            calibration.Calibration, "_closest_point_fit"
+        ) as mock_closest_point_fit:
+            mock_extracted_data.return_value = no_motion_data
+            mock_closest_point_fit.return_value = linear_transformation
+            calibrator = calibration.Calibration()
+
+            result = calibrator._calibrate(dummy_measure)
+
+            assert result["scale"] == linear_transformation["scale"]
+            assert result["offset"] == linear_transformation["offset"]
+
+
+@pytest.mark.parametrize(
+    "chunk_num, expected_len",
+    [(0, (3600 * 72)), ((1), (3600 * 84)), ((2), (3600 * 84))],
```

Comment:
So when I create the dummy data below it's only 84  hours, so that tests the senario where the chunk gets called, it goes out of range, and just returns the whole thing. 

Because right now the chunk size is hard coded to 12 hours, if the number of hours isnt divisible by 12, the last chunk will be some random number of hours less than 12. So when that last chunk is added, it'll be out of range, so it should just return the whole thing at that point. Hence Chunk size = 1 and Chunk size = 2 should both return 84 hours. 

Thinking it out maybe it's better to have the duration_hours in the assembly be something like 86 hours to make that clearer.